Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Tools/Notifications.hs

```diff
@@ -81,8 +96,68 @@ data UpdateLocationNotificationReq = UpdateLocationNotificationReq
   }
   deriving (Generic, ToJSON, FromJSON, ToSchema, Show)
 
-templateText :: Text -> Text
-templateText txt = "{#" <> txt <> "#}"
+data FCMReq = FCMReq
+  { notificationKey :: Text,
+    subCategory :: Maybe Notification.SubCategory,
+    showType :: FCM.FCMShowNotification,
+    entityId :: Text,
+    entityType :: FCM.FCMEntityType,
+    sound :: Maybe Text,
+    overlayReq :: Maybe FCM.FCMOverlayReq,
+    priority :: FCM.FCMAndroidMessagePriority
+  }
+
+instance Default FCMReq where
+  def =
+    FCMReq
+      { notificationKey = mempty,
+        subCategory = Nothing,
+        showType = FCM.SHOW,
+        entityId = mempty,
+        entityType = FCM.Product,
+        sound = Nothing,
+        overlayReq = Nothing,
+        priority = FCM.HIGH
+      }
+
+defaultFCMReq :: FCMReq
+defaultFCMReq = def
```

Comment:
@ErasedSoul Seems like notificationKey, entityId and entityType always needs to be added explicitly by Developer so Instead of making it Abstract by using default we can create a function like below that should be used to create FCM payload so that the Dev is aware that this fields should be added properly and rest all fields even with default it should work fine so those can be updated using `modifier` if required.

```
createFCMReq :: Text -> Text -> FCM.FCMEntityType -> (FCMReq -> FCMReq) -> FCMReq
createFCMReq notificationKey entityId entityType modifier 
 = modifier def { .. }
```